ocaml/typing/env.mli

```diff
@@ -36,7 +36,7 @@ type summary =
   | Env_modtype of summary * Ident.t * modtype_declaration
   | Env_class of summary * Ident.t * class_declaration
   | Env_cltype of summary * Ident.t * class_type_declaration
-  | Env_open of summary * Path.t
+  | Env_open of summary * Longident.t
```

Comment:
I don't like that the environment you get out will not always be exactly the same now if e.g. the available `cmi` files have changed. I'd probably prefer that this stay as a `Path.t` and include the locks data.

Also, it loks like we're not storing locks in this `summary`, which is definitely a (preexisting) bug.

Reply:
Adding locks to summary is tracked by internal ticket 1778.

Without doing that, `Env_summary` ruins locks anyway, I think for now `Env_open` can just contain `Path.t` without `lock_item list`. I will leave a CR to fix it in the future.